Airtable oauth migration (#9687)
PipedreamHQ/pipedream: docs/docs/airtable/oauth-migration-2024-02/README.md

- Search for Airtable and connect your account
- This newer Pipedream integration uses OAuth instead of an API Key

![Airtable Account Connection](https://res.cloudinary.com/dpenc2lit/image/upload/v1704392183/Screenshot_2024-01-04_at_10.16.12_AM_le364k.png)

You can determine which workflows are connected to the legacy Airtable (API Key) app by expanding the account row on your Airtable (API Key) account connection.
![Airtable Accounts](https://res.cloudinary.com/dpenc2lit/image/upload/v1704347928/Screenshot_2024-01-03_at_9.58.43_PM_haaqlb.png)

2. **Update Your Workflows**: 

- After reconnecting to Airtable via OAuth, you'll need to update your existing workflows that use the legacy Airtable app.
- Remove any legacy Airtable sources and re-add the source using the new Airtable app
- Remove any legacy Airtable actions and re-add them using the new Airtable app

<br/>

3. **If you're using Airtable in code:**

- Change any of your code steps to reference `airtable_oauth` instead of `airtable`. 
- Modify your authorization headers accordingly

In Node.js, you would modify your authorization header like this:

    `"Authorization": ${this.airtable.$auth.api_key}`

    to 

>     ```Authorization: `Bearer ${this.airtable_oauth.$auth.oauth_access_token}` ```

Reviewer: you probably have to escape the backticks that wrap Bearer, like so:
```suggestion
    `Authorization: \`Bearer ${this.airtable_oauth.$auth.oauth_access_token}\``
```

does that work?

Author: This seemed to work:

``` Authorization: `Bearer ${this.airtable_oauth.$auth.oauth_access_token}` ```

Would that be acceptable? In preview, it looks good.

Reviewer: this is what I see 
![Screenshot by Dropbox Capture](https://github.com/PipedreamHQ/pipedream/assets/20780753/770b42b4-f422-4e33-94ad-4aff5c9f3672)

Author: Sorry, I made some changes to lessen the indent and it worked. Was having issues committing my change, but now it should be working!

# Update to the Airtable Integration on Pipedream (January 2024)

Effective February 1st 2024, Airtable's API Key authentication method will be deprecated. To learn more about this change, please visit Airtable’s [dedicated support page](https://support.airtable.com/docs/airtable-api-key-deprecation-notice).

### How will this impact my workflows?
Starting February 1st 2024, all Pipedream steps using the legacy Airtable (API Key) integration including triggers and actions will no longer be able to authenticate with Airtable. 

### What do I need to do?

> 

Reviewer: I'd add another linebreak here. If just a newline doesn't do it, you should be able to add a `<br>` I believe.

1. **Reconnect your Airtable account**: Visit [https://pipedream.com/accounts](https://pipedream.com/accounts) and search for Airtable and connect your account. This newer Pipedream integration uses OAuth instead of an API Key. 

![Airtable Account Connection](https://res.cloudinary.com/dpenc2lit/image/upload/v1704326732/Screenshot_2024-01-03_at_4.02.24_PM_kvasnc.png)

> You can determine which workflows are connected to the legacy Airtable (API Key) app by expanding the account row.

Reviewer: I'd maybe just clarify that users will need to expand the row for the API key account, not the new one they've connected in step 1.

![Airtable Accounts](https://res.cloudinary.com/dpenc2lit/image/upload/v1704347928/Screenshot_2024-01-03_at_9.58.43_PM_haaqlb.png)

> 2. **Update Your Workflows**: After reconnecting to Airtable via OAuth, you'll need to update your existing workflows that utilize Airtable:

Reviewer: Then similarly to the above suggestion, maybe something like this:

```suggestion
2. **Update Your Workflows**:
-  After reconnecting to Airtable via OAuth, you'll need to update your existing workflows that use the legacy Airtable app.
- Remove any legacy Airtable sources and re-add the source using the new Airtable app
- Remove any legacy Airtable actions and re-add them using the new Airtable app
```

    - Remove your old Airtable triggers, and reconfigure them using the new Airtable app.
    - Remove your old Airtable action, and reconfigure them using the new Airtable app.

> 3. **If you are using code steps:**

Reviewer: ```suggestion
3. **If you're using Airtable in code:**
```

    - Change any of your code steps to reference `airtable_oauth` instead of `airtable`. 

>     - Modify your authorization headers accordingly, from

Reviewer: Maybe overkill, but might be helpful to mention the python syntax as well

Author: added in Python syntax as well

# Update to the Airtable Integration on Pipedream (January 2024)

> Effective 2024-02-01, Airtable will be deprecating its API Key authentication method. To learn more about this change, please visit Airtable’s [support page](https://support.airtable.com/docs/airtable-api-key-deprecation-notice).

Reviewer 2: ```suggestion
Effective February 1st 2024, Airtable's API Key authentication method will be deprecated. To learn more about this change, please visit Airtable’s [dedicated support page](https://support.airtable.com/docs/airtable-api-key-deprecation-notice).
```

> ### Why is Airtable deprecating API Key support?

Reviewer: IMO you can drop this section

Author: Dropped.

OAuth provides more granular control of resources and scopes, and enables both developers and end-users to extend Airtable while ensuring the highest grade of security.

### How will this impact my workflows?

> Starting 2024-02-01, any Pipedream workflow using the legacy Airtable (API Key) integration will no longer work. 

Reviewer 2: ```suggestion
Starting February 1st 2024, all Pipedream steps using the legacy Airtable (API Key) integration including actions and trigger will no longer be able to authenticate with Airtable. 
```

### What do I need to do?

> 1. **Reconnect your Airtable account**: Visit https://pipedream.com/accounts and connect your Airtable account using the new application, which utilizes OAuth.

Reviewer: make sure to wrap the url as a link so it's clickable. I'd also say something like, "...search for Airtable and connect your account. This newer Pipedream integration uses OAuth instead of an API Key.", or something.

![Airtable Account Connection](https://res.cloudinary.com/dpenc2lit/image/upload/v1704326732/Screenshot_2024-01-03_at_4.02.24_PM_kvasnc.png)

> You will be prompted to grant Pipedream access to Airtable, and you can choose an individual base in a workspace, all current and future bases in a given workspace, or all workspaces.

Reviewer: I don't think you need to spell this out

Author: Dropped this section as well for brevity.

> <img src="https://res.cloudinary.com/dpenc2lit/image/upload/v1704326731/Screenshot_2024-01-03_at_4.02.48_PM_mnukaj.png" width=400>

Reviewer 2: ```suggestion
<img alt="Configuring Airtable accounts to use the OAuth authentication instead of API authentication within Pipedream" src="https://res.cloudinary.com/dpenc2lit/image/upload/v1704326731/Screenshot_2024-01-03_at_4.02.48_PM_mnukaj.png" width=400>
```

Author: I've dropped this section for brevity.

> 2. **Update Your Workflows**: After reconnecting to Airtable via OAuth, you'll need to update your existing workflows that utilize Airtable:

Reviewer: probably worth calling out that you can see which workflows are using Airtable from the account row
![Screenshot by Dropbox Capture](https://github.com/PipedreamHQ/pipedream/assets/20780753/1fa2e7c3-cfc1-4951-a147-3619b405739f)

Author: good suggestion, added

    - Remove your old Airtable triggers, and reconfigure them using the new Airtable app.
    - Remove your old Airtable action, and reconfigure them using the new Airtable app.

> 2a. **If you are using code steps:**

Reviewer: nit: "2a" is kind of awkward, especially if there's no "2b"

Author: changed to 3.

    - Change any of your code steps to reference `airtable_oauth` instead of `airtable`. 
    - Modify your authorization headers accordingly, from

>     ```"Authorization": `${this.airtable.$auth.api_key}`,```

Reviewer 2: ```suggestion
   `"Authorization": `${this.airtable.$auth.api_key}`,`
```

Four back ticks is usually for multi-line code blocks and where the language is known and syntax highlighting added. Here it's not clear if it's for curl, Node.js or Python.

Author: Thanks for the suggestion - added.

    to 

>     ```Authorization: `Bearer ${this.airtable_oauth.$auth.oauth_access_token}```

Reviewer 2: ```suggestion
    `Authorization: `Bearer ${this.airtable_oauth.$auth.oauth_access_token}`
```

Same suggestion as above.

Author: I wasn't able to get this exactly with single backticks.
